source/slang/slang.cpp

```diff
@@ -3825,46 +3824,44 @@ CompositeComponentType::CompositeComponentType(
     }
 }
 
-SlangResult CompositeComponentType::computeDependencyBasedHash(slang::Checksum* outHashCode)
+void CompositeComponentType::updateDependencyBasedHash(
+    HashBuilder& builder,
+    SlangInt entryPointIndex)
 {
-    auto componentCount = getChildComponentCount();
+    auto hashGen = builder.hashGen;
+    auto context = &builder.context;
 
-    slang::Checksum hashCode;
-    MD5HashGen hashGen;
-    MD5Context context;
-    hashGen.init(&context);
+    auto componentCount = getChildComponentCount();
 
-    for (Index i = 0; i < componentCount; ++i)
+    // Add file path dependencies to the hash - all child components
```

Comment:
`fileDeps`, and entry point names should be hashed in `ComponentType`, not here.